### ISSUE TITLE
open link in new page

### ISSUE THREAD

**Issue by alimd** [mojombo/github-flavored-markdown#28]
how can add target="_blank" to markdown links `[test](http://test.com)` in github?!

**Comment by skeggse** [comment]
It seems unlikely that this will ever become a feature of markdown or gfm.

That said, you can use standard HTML to add a link that targets a new window/tab:

`<a href="http://example.com" target="_blank">http://example.com</a>`

<a href="http://example.com" target="_blank">http://example.com</a>

**Comment by trliner** [comment]
Hmm, your example link rendered as:

``` html
<a href="http://example.com" target="_blank"></a>
<a href="http://example.com">http://example.com</a>
```

So clicking the visible link doesn't actually open the URL in a new window. :disappointed: 

However, it does work properly if the body of the link isn't a full URL:
<a href='http://www.example.com' target='_blank'>example</a>

**Comment by skeggse** [comment]
> However, it does work properly if the body of the link isn't a full URL:
> <a href="http://example.com" target="_blank">example</a>

Ahh, interesting. I guess that makes sense.

**Comment by paulmenzel** [comment]
Would adapting the “Kramdown extension” to use [[link](URL){options}](http://kramdown.gettalong.org/syntax.html) be an option?

**Comment by alimd** [comment]
@paulmenzel now working in github

**Comment by trliner** [comment]
@AliMD I'm having a hard time getting that to work. What am I doing wrong?

`[test](http://test.com){:target="_blank"}` renders as    [test](http://test.com){:target="_blank"}

**Comment by chennighan** [comment]
Same @trliner, from what I've read, multiple people have had to revert that fix because github flavor markdown doesn't support it.

**Comment by RichardHightower** [comment]
![test](https://github.com/mojombo/github-flavored-markdown/issues/28)

**Comment by DenysVuika** [comment]
The issue is still open since 23 Nov 2012

**Comment by gautamkrishnar** [comment]
Looks like it is not working...
```
<a href="http://stackoverflow.com" target="_blank">Go</a>
```

**Comment by datapolitan** [comment]
Same problem with my [README](https://github.com/Datapolitan-Training/data-vis-excel/blob/gh-pages/README.md). The raw Markdown has the HTML but it's being scrubbed off somehow when it's rendered.

**Comment by exexzian** [comment]
I wonder how come this simple feature issue is open since **2012** _(4+ years seriously???)_. 

And before finding out this issue being reported here, I was giving syntax check again and again to kramdown ...

Wish Github starts supporting this feature

**Comment by Nerajno** [comment]
So its now August and I am gonna keep talking about this. Why is there no solution to this ?
@github

**Comment by 0xFFrancesco** [comment]
Same here, even entering plain HTML with `target='_blank'` set does not work.

**Comment by grv87** [comment]
Github doesn't implement new custom extensions to Markdown format and looks towards CommonMark. I think that https://github.com/commonmark/CommonMark/pull/105 should be approved and merged into CommonMark specification first, then Github would implement this common specification in GFM.

**Comment by seneca-wolf** [comment]
Wow, it blows my mind that there isn't a simple way to do this. Who doesn't find it annoying when tabs open in the same window? That should not even be the default behavior. Worse, it is apparently something that can't even be remedied right now. Sad.

**Comment by SvenJedeck** [comment]
This is really sad

**Comment by redhatstuart** [comment]
Also have a need for this.

**Comment by coderberry** [comment]
Same here

**Comment by lucasdavid** [comment]
[a nice link called link](http://stackoverflow.com){:target="_blank"}?

**Comment by juanroldan1989** [comment]
Ditto

**Comment by gabrielnieves18** [comment]
+1

**Comment by elijah-io** [comment]
Bump.

**Comment by Zarquan** [comment]
+1

**Comment by ghost** [comment]
holy heck, guys, github leaves dis issue for SO LONG! whoa! 
plus, I have the same problem

**Comment by JamesAndersonJr** [comment]
+1

**Comment by Marcial1234** [comment]
+1

**Comment by albertpeiro** [comment]
+1

**Comment by jamesjwilson** [comment]
+1

**Comment by brainthinks** [comment]
+1

**Comment by 174n** [comment]
+1

**Comment by nirantak** [comment]
+1

**Comment by johba37** [comment]
+1

**Comment by wldhg** [comment]
+1

**Comment by VolkmarR** [comment]
+1

**Comment by brgaines** [comment]
+1

**Comment by SkyHustle** [comment]
still waiting on this....

**Comment by nullsibnelf** [comment]
why? this is as old as the whole www..... 
Why is implementing a target such a problem?

**Comment by kelvinau** [comment]
+1

**Comment by jasonmimick** [comment]
+1

**Comment by TakingItCasual** [comment]
+1

**Comment by jayliu50** [comment]
+1

**Comment by Michael-Schaer** [comment]
+1

**Comment by roberto910907** [comment]
+1

**Comment by nuhkoca** [comment]
+1

**Comment by joelzamboni** [comment]
+1

**Comment by zoetian** [comment]
+1

**Comment by shushugah** [comment]
+1

**Comment by Kaisky** [comment]
+1

**Comment by NetSensors** [comment]
+1

**Comment by IeiuniumLux** [comment]
{:+1="_one_more_year"}

**Comment by CaptainDredge** [comment]
+1

**Comment by jonsgity** [comment]
+1

**Comment by fathertime100** [comment]
+1

**Comment by NickLiffen** [comment]
+1

**Comment by r3faat1** [comment]
+1... lol.

**Comment by jvtao21** [comment]
+1

**Comment by msteger93** [comment]
+1

**Comment by mmajcica** [comment]
+1

**Comment by ebeltran1981** [comment]
+1

**Comment by KevinNovak** [comment]
+1

**Comment by ihtigum** [comment]
+1

**Comment by aravindvnair99** [comment]
+1

This has long been overdue.

@github Any updates?

**Comment by juniorcesarabreu** [comment]
+1

**Comment by markesejd** [comment]
+1

**Comment by CodeDmitry** [comment]
+1

**Comment by domrod** [comment]
+1

**Comment by TheBestPessimist** [comment]
+1

**Comment by AlexWayfer** [comment]
Users, please stop write `+1`. Use reactions (Emojis) at first post.

Collaborators, please lock this conversation.

**Comment by exodus4d** [comment]
+1

**Comment by KevinNovak** [comment]
Is this being seriously considered? Leaving a project page is bad user experience in most cases and just involves going back and forth. At least having the option would be nice.

**Comment by jpcmf** [comment]
+1

**Comment by DanSilva41** [comment]
+1

**Comment by madison-codes** [comment]
+1

**Comment by seongbae** [comment]
+1

**Comment by vladmorosanu97** [comment]
+1

**Comment by riomyers** [comment]
+1 - I'm surprised opening links in a new window isn't already a built-in feature, this is a basic function that should be implemented

**Comment by starikcetin** [comment]
+1

**Comment by Harti** [comment]
+1

**Comment by aleshchev** [comment]
+1

**Comment by rjouhann** [comment]
+1

**Comment by RafaelBroseghini** [comment]
+1 - I'm surprised opening links in a new window isn't already a built-in feature, this is a basic function that should be implemented

**Comment by roberto910907** [comment]
How many +1 do we need to implement this feature?

**Comment by starikcetin** [comment]
Looks like @AlexWayfer took their sweet time to downvote all the comments. Impressive to be honest.

**Comment by AlexWayfer** [comment]
@starikcetin Only "+1" unhelpful comments. I've asked to not write this… rubbish, and rate the first comment (issue starting).

**Comment by shinebayar-g** [comment]
I don't want to someone leaves my readme/wiki page back and forth while navigating. It's actually very sad to see no one gave a fuck on this issue. :-1:

**Comment by AlexWayfer** [comment]
> I don't want to someone leaves my readme/wiki page back and forth while navigating.

Use HTML `<a>` tag.

**Comment by shinebayar-g** [comment]
> Use HTML `<a>` tag.

You mean this?
`<a href="https://www.google.com/" target="_blank">google.com</a>`

Not working.

**Comment by ttodua** [comment]
+1

**Comment by Makoehle** [comment]
++1
2 plus signs to check if AlexWayfer has a bot in the background or is srsly doing this manually ^^

**Comment by ttodua** [comment]
as in many cases, the devs doesnt mind our +1 or upvotes.
if they wanted to do that, they could have done in 2012, in several days after opening this topic, because it doesn't need much thinking, that this feature will be needed by many people. 
just, they say NO.  
It will be a bad more sign and unprofessionalism of devs, if they fixed this problem at this moment (after 6 years of raising the issue). so, i suppose this is "won't fix".

**Comment by AlexWayfer** [comment]
> 2 plus signs to check if AlexWayfer has a bot in the background or is srsly doing this manually ^^

I don't want to waste time for a bot.

I'm subscribed to this issue (via button at the right menu), and I'm receiving many useless `+1` in my email, and I don't want to receive them. I've asked about this, but people don't want to hear others.

-----

I was wrong with `<a>` tag suggestion, I didn't think that GitHub cuts attributes from tags. My bad.

**Comment by guylepage3** [comment]
+1

**Comment by Harti** [comment]
@AlexWayfer 
![image](https://user-images.githubusercontent.com/2056108/42769458-79e8c7cc-8922-11e8-8f1a-677efb331a4d.png)

**Comment by AlexWayfer** [comment]
@Harti, I want to get updates of this issue. `+1` is not update.

**Comment by Harti** [comment]
Just set yourself a reminder then, every year. Put the link in the reminder‘s notes, or bookmark the issue.

Chances are this will still be open and unresolved every time you view it.

But getting salty because other people are using this platform to nag the developers to tackle this actually important feature, and are disturbing your inbox peace, is ridiculously childish and entitled.

**Comment by AlexWayfer** [comment]
> Just set yourself a reminder then, every year. Put the link in the reminder‘s notes, or bookmark the issue.

Just don't write `+1`. It's pointless.

Instead of every interested person like me changing subscription to issue to a reminder, I'm trying to change the whole situation, make public issue discussions more informative. One situation, one rule, instead of many moves of many people.

> But getting salty

I'm just trying to make this place better, without aggression.

> because other people are using this platform to nag the developers to tackle this actually important feature

No, `+1` doesn't nag the developers. If they will want to do something from issues of this projects — they'll use sorting by 👍, I'm pretty sure. Look, this issue is already the first: https://github.com/mojombo/github-flavored-markdown/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

And `+1` comments don't help for this sorting, nor don't nag the developers, nor don't give some useful information for all 80 participants of this issue. 

So, please, use 👍 button at the issue-starting comment, subscribe if you want to watch for this issue, and be free. `+1` comments are obsolete, before emotions buttons feature. Don't litter emails of dozens of people with absolute trash like `+1` comments.

**Comment by jamesjwilson** [comment]
Just unsubscribe from it.  You're creating more spam, thus making me truly wonder if alleviating spam is truly your desire rather than just a simple pet peeve that people aren't using the system the way you desire it to be used.

**Comment by AlexWayfer** [comment]
> Just unsubscribe from it.

No.

> You're creating more spam

No.

**Comment by jamesjwilson** [comment]
> No.

Yes.

> No.

Yes.

**Comment by AlexWayfer** [comment]
> Yes.

I'll not.

> Yes.

Just count before you claim in order not to be a liar.

**Comment by jamesjwilson** [comment]
I just wanted to prove that you had no desire to clean up spam.  Thank you for making that way easier than I anticipated.  Now we know we can just ignore your nonsense and have a clear conscious of doing so.

**Comment by m52go** [comment]
+1

GITHUB Y U NO

<img src="https://memegenerator.net/img/images/166088/y-u-no.jpg" width="250" />

Clippy, help?

**Comment by ani10030** [comment]
+1
I bit of my contribution to the ongoing spam 😜

**Comment by tusharshuklaa** [comment]
+1
And a bit of my contribution as well 😎👀

**Comment by joshyap** [comment]
+1
I'm getting involved in Hacktoberfest right now and wanted to update some particularly link-heavy open source documentation, only to stumble across this.

**Comment by ttodua** [comment]
posting, commenting or upvoting here has no meaning. 
When companies want to make something, that enough even if 1 user told that feature-request. If a company was to anknowledge that, it was enough. If company ignores user request for some reason, then it doesnt have any meaning if even 500 people posted here. 

see, the Original request was 2012 yr. Company ignored the request and that was all. This is a politic you will see everywhere, when users request a simplest change, but companies just ignore that immedately, and no further "upvoting", even of hundreds of people, change anything.

**Comment by starikcetin** [comment]
Of course it has a meaning. It means more and more people want this feature. Implementing or not implementing this feature has nothing to do with that.

**Comment by bdeleonardis1** [comment]
This is actually ridiculous that they can't spend an hour adding this feature.

**Comment by MichaelKim0407** [comment]
Just had the same problem and Google led me here.. For everyone that says `+1`, I don't think this is the correct repo?

**Comment by aravindvnair99** [comment]
> Just had the same problem and Google led me here.. For everyone that says `+1`, I don't think this is the correct repo?

Yes, I too was thinking the same. Because the organisation profile of GitHub is this: https://github.com/github and I could find this: https://github.com/github/markup although I'm not really sure of this.

**Comment by starikcetin** [comment]
I opened this issue in the repo @aravindvnair99 mentioned: https://github.com/github/markup/issues/1243

**Comment by joenano** [comment]
AI will fix this before github, so about 2000 years from now we should have the functionality.

**Comment by danon** [comment]
Any chance this will be implemented?

**Comment by siddharthjthapa** [comment]
Fix this issue github

**Comment by starikcetin** [comment]
Hello people, this is the response I received from the GitHub markup repository:

> 👋 Please reach out [to our support team](https://github.com/contact) with your feature request — this repository is just for issues for the github-markup gem, which doesn't actually perform Markdown transformation; it just selects a library. We don't support this at all right now, so by directing this to the support team, it helps keep track of feature requests and how many people are asking for them. Thanks!

https://github.com/github/markup/issues/1243

They want us to send support tickets.

**Comment by frosty939** [comment]
how is this STILL not implemented..

**Comment by yazeedalrubyli** [comment]
R.I.P
2012-2019

**Comment by dmorenogogoleva** [comment]
:octocat: same problem

**Comment by SorourMo** [comment]
Seriously?! Why no action? We need this.

**Comment by th3uiguy** [comment]
Would love to see this feature

**Comment by ThBastos** [comment]
R.I.P
11-23-2012 > 02-12-2019

**Comment by gitnepal** [comment]
2019 .....

**Comment by danielquijada** [comment]
Read through this thread expecting to find somewhere a "Hey, they finally implemented it, you just need to do <whatever stuff you need to do>"

But nope.

2320 days ago.

**Comment by serkansahin** [comment]
March 2019 we're still waiting for this Github...

**Comment by Mararsh** [comment]
I need this... I searched and tried solutions, but all failed.
Please fix this. Thanks!

**Comment by edfilbasan** [comment]
Doesn't make sense how this hasn't been resolved. This can be handed off to a summer intern...

**Comment by uclaeagit** [comment]
This is unbearable when you link to a site that does a redirect and you can't easily get back to your readme.

Microsoft should put all these unresolved github feature requests in a hat and at least pick a few to work on each month - to give us some hope.

**Comment by vpzomtrrfrt** [comment]
All these comments are implying that this is an obvious feature that should exist, but I don't get it.  I can't imagine any case where this should be allowed

**Comment by AlexWayfer** [comment]
> I can't imagine any case where this should be allowed

Badge of tests, version, code quality or whatever with the link to the page of a project on this service.

**Comment by jpbochi** [comment]
> I can't imagine any case where this should be allowed

For example, the references section of basically every wikipedia page is full of "external" links decorated with ![](https://en.wikipedia.org/w/skins/Vector/images/external-link-ltr-icon.png). There's no reason why a README page could not follow a similar pattern. And a nice implementation by GitHub could even enforce that every such links are decorated with the same icon, so that no one gets surprised by that.

**Comment by uclaeagit** [comment]
We have private repos. Our READMEs link to the AWS console for that item. AWS does some kind of redirect, so even clicking back doesn't get you back to the README you were looking at. You have to click back really fast, or use the arrow/history.

**Comment by TheBestPessimist** [comment]
Until [CommonMark](https://commonmark.org/) adds this feature to the standard, middle click is your friend folks. (or control click if you are so inclined).

**Comment by 174n** [comment]
... gitlab is your friend folks. (or bitbucket if you are so inclined).

**Comment by TheBestPessimist** [comment]
I have always found [this](https://docs.gitlab.com/ee/administration/operations/unicorn.html#unicorn-worker-killer) to be a "No thank you, GitLab".

**Comment by AlexWayfer** [comment]
> Until CommonMark adds this feature to the standard

There is a discussion: https://talk.commonmark.org/t/consistent-attribute-syntax/272

---

> I have always found this to be a "No thank you, GitLab".

Do you think that this case is better in GitHub? Oh, wait, GitHub has no configuration options, because it's not self-hosted. So… there are pros and cons, and this is off-topic for this issue.

**Comment by uclaeagit** [comment]
We had a couple "No thank you, Gitlab"s. The main one is we didn't want to use their onsite product, and their hosted product seemed like kind of an after-thought to that. IE - there was no concept of an organization for your private repos. Each private repo you pick who gets access - **from everyone on gitlab**. It would be so easy to accidentally pick the wrong person. Also the documentation was really confusing between the onsite and hosted version.

My only serious gripe with github.com - is no folders! Argh. We have to resort to brittle, awkward naming conventions to keep our repos organized. And because we have AWS webhooks into all of them - renaming breaks everything.

**Comment by aravindvnair99** [comment]
Is this even the right place to make GitHub aware of this issue? The issue has been open since 2012 and the last commit to the repo was in 2010. I'm wondering if this is the right repo in the first place.

**Comment by akaia-shadowfox** [comment]
+1

**Comment by ccarcaci** [comment]
+1

**Comment by alan23273850** [comment]
What's going on??!! still no progress??!

**Comment by sabuhish** [comment]
2020 still no progress :(

**Comment by roberto910907** [comment]
> AlexWayfer commented on Jul 16, 2018
> Harti, I want to get updates of this issue. `+1` is not update.

@AlexWayfer wish you the best of luck with that LOL.

**Comment by pufferbatterie** [comment]
+1

**Comment by haugthom** [comment]
+100

**Comment by agavrel** [comment]
## **[+1](https://github.com/mojombo/github-flavored-markdown/issues/28)**

**Comment by green-green-avk** [comment]
+1

**Comment by wokuang** [comment]
+1

**Comment by KTibow** [comment]
Cheater method with image tag: you can open the same page in a new tab, it seems...
- ```
  ![New tab]()  
  ```
  <s>![New tab]()</s>(Doesn't work now)  

This takes you to the same page in a new tab. If you try using `http`, or `https`, then GitHub Camo gives you an error:
- ```
  ![New tab](http://example.com/)  
  ![New tab](https://google.com/)  
  ```
  ![New tab](http://example.com/)  
  ![New tab](https://google.com/)  

These both give you the error message `Non-Image content-type returned`.  
How about `//`?
- ```
  ![New tab](//example.com/)  
  ```
  <s>![New tab](//example.com/)</s>Doesn't work anymore  

That opens the same page in a new tab, oddly. The same goes for any other custom protocol.  
You can use relative links to open in a new tab. This might be handy with GitHub Pages:
- ```
  ![New tab](/)  
  ![New tab](testing123)  
  ![New tab](/mojombo)  
  ![New tab](../)  
  ![New tab](../issues/28?foo=bar)  
  ```
  ![New tab](/)  
  <s>![New tab](testing123)</s>Doesn't work (filters by author:testing123 in issues)  
  ![New tab](/mojombo)  
  <s>![New tab](../)</s>Works, but not in a new tab  
  <s>![New tab](../issues/28?foo=bar)</s>Works, but without a new tab  

## Recap:
- You can use image links with the image path set to the URL you want to open.
- GitHub Camo can stop you from doing this in some cases.
- You can use relative links (`/`, `/something`, <s>`../`, `anotherthing`</s>)
- <s>Putting nothing for the URL makes the same page open in a new tab (`![New tab]()`)</s>

Until they re-implement anything, or add a redirect scheme on the domain `github.com`, we're restricted to the same page, in a new tab.

**Comment by Dn-a** [comment]
it's crazy!
Does Internet Explorer have anything to do with it?

**Comment by KTibow** [comment]
> Does Internet Explorer have anything to do with it?
@Dn-a what do you mean? The markdown is sent to GitHub's markdown parser, which converts it to HTML, which the browser then renders. In the end, it's plain old HTML. In the start, it's just markdown.

**Comment by Dn-a** [comment]
@KTibow I was ironic 😄

**Comment by hazem3500** [comment]
Truly sad.

**Comment by KTibow** [comment]
Side note @uclaeagit: GitHub has folders. When you create a file, just type "folder/folder/file" or whatever. It won't allow empty folders though.

**Comment by dpolivaev** [comment]
There is a workaround working only on GitHub pages, see https://stackoverflow.com/a/63546509/1833472

**Comment by danon** [comment]
@dpolivaev Also doesn't work in IntelliJ (not sure about VS Code).

**Comment by dpolivaev** [comment]
Yes, as I wrote, it requires another markdown processor. But for making sites it is a lifesaver.

**Comment by KTibow** [comment]
@mojombo isn't even watching this repo in the watchers list.

**Comment by bhavyagosai** [comment]
Still open since 2012 now this is depressing af 😔 GITHUB bruh do something

**Comment by KTibow** [comment]
@github come on...

**Comment by dpolivaev** [comment]
I noticed that the last commit in this repository was done 12 years ago. So I think it is just the wrong place for this issue.
I think that https://github.com/github/cmark-gfm/issues could be the right one.

How do you see it, folks?

**Comment by guptarohit** [comment]
@github come on, it 2021 now... :confused:

**Comment by vvaltchev** [comment]
Any chance to make target="_blank" to work? It would be really useful.

**Comment by Nufflee** [comment]
almost 8.5 years and counting..

**Comment by v-vovk** [comment]
hello from February 2021)

**Comment by devChrisho** [comment]
+1

**Comment by namishkhanna** [comment]
@AliMD @mojombo Resolved the Issue

**Try This:** https://namishkhanna.github.io/newtab/?url=http://test.com

**For Reference:** https://github.com/namishkhanna/newtab

**Comment by MinCheTsai** [comment]
need...

**Comment by AbshkPskr** [comment]
when ?

**Comment by moovs** [comment]
anyone still waiting???

**Comment by Tecktron** [comment]
Still wanting and waiting, 3,107 days and counting

**Comment by 174n** [comment]
Even gitlab doesn't have that I think. So maybe it's some kind of security issue which prevents them from implementing it

**Comment by AlexWayfer** [comment]
> Even gitlab doesn't have that I think. So maybe it's some kind of security issue which prevents them from implementing it

I guess you're right, there were security issues, check for example [a note on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). And there is still no support of `rel="noopener"` in some browsers like IE, Opera, Samsung Browser, etc.

**Comment by asmanp2012** [comment]
+1

**Comment by dhvakr** [comment]
Sad thing i've seen is, the issue is still open **since 23 Nov 2012** .. hmmm

**Comment by alimd** [comment]
> Sad thing i've seen is, the issue is still open **since 23 Nov 2012** .. hmmm

Write new post with `+1` may be help 😆

**Comment by Wyndace** [comment]
+1

**Comment by Md-Kais** [comment]
we need a solution for this 
[go](http://stackoverflow.com){:target="_blank"}
Can anyone help me?

**Comment by Liverrrrr** [comment]
That only works in [Kramdown](http://kramdown.gettalong.org/syntax.html). They should make a way to render md files with it in github

**Comment by Liverrrrr** [comment]
+1

**Comment by jonathancbrito** [comment]
+1

**Comment by wakoliVotes** [comment]
Hi, community,

Any solution since 2012? ` **taget="_blank**"` not working at all. Just opens on the same GitHub page.

**Comment by Gotterbild** [comment]
+1
Need `target="_blank"` working at GitHub repo Wiki.

It steals a lot of time of our team.

**Comment by 174n** [comment]
> It steals a lot of time of our team.

use middleclick to open in a new tab

**Comment by Gotterbild** [comment]
@174n 
> > It steals a lot of time of our team.
> 
> use middleclick to open in a new tab

Dude, no offence, but do you think I'm stupid? 🙂

1. Not everyone uses mouse. Some people use touchpads, some use touch screens (phones, tablets)
2. Even with mouse middle click or touchpad triple tap you have to do two additional actions:
   1. Aim mouse to tab head
   2. Perform additional clicks
   
   Alternatively, you can use keyboard, but that would require two-three actions:
   1. Put hand on keyboard (sometimes it's not there)
   2. Put fingers on Ctrl+Tab (not sure on Mac combination)
   3. Push buttons
   
The higher amount of actions to be performed, the higher probability of error is. So, those additional actions increase probability of errors roughly 100-150%.

And I don't mention persons with disabilities...

When you need to do additional actions once, it's not a problem. But when you need to do it thousands times per day, those redundant fractions of seconds turn into minutes. And then you multiply that on amount of users.

Hope you get the point.

**Comment by alimd** [comment]
This is very simple request and I don't understand this heavy resistance!

**Comment by njfamirm** [comment]
+1

**Comment by snoop2head** [comment]
Happy new year guys! Still waiting though 😅

**Comment by krisalyssa** [comment]
I can live with GFM not providing a way to add a `target` attribute to the `a` tag. But what really chaps me is that if I write a raw HTML link like `<a href="foo" target="_blank">...</a>`, *GFM strips the `target` attribute*.

If I'm going to the trouble of including raw HTML in my document, I'd like it to be passed through untouched, please. If there's some kind of security risk, and I can't be assumed to understand and accept the risk, then this behavior *should be documented* in [the GFM spec](https://github.github.com/gfm/#html-tag).

**Comment by krisalyssa** [comment]
A coworker of mine found a blog post (not from GitHub) which describes how `target="_blank"` can be a security and performance issue: https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/

If this is GitHub's rationale, great. It should be documented so this issue can be closed *after nine years*.

**Comment by filipmu** [comment]
Here is way to add a little reminder to press the middle mouse button on a link to open it in a new window and a way to bring more attention to this issue yourself:

<a href="http://example.com" target="_blank" rel="noopener noreferrer" title="***Please middle click on mouse to open in new tab***&#013Github Markdown does not support opening new pages (github.com/mojombo/github-flavored-markdown/issues/28)">example.com</a>

```<a href="http://example.com" target="_blank" rel="noopener noreferrer" title="***Please middle click on mouse to open in new tab***&#013Github Markdown does not support opening new pages (github.com/mojombo/github-flavored-markdown/issues/28)">example.com</a>```

**Comment by ramchandra-shahi** [comment]
Following this thread, for update.

**Comment by AlexWayfer** [comment]
> Following this thread, for update.

![image](https://user-images.githubusercontent.com/6510020/153180440-1a360bbd-b652-4e30-9b05-8af1a5970586.png)

**Comment by KTibow** [comment]
The post that @CraigCottingham linked seems a little outdated. These days, `rel="noopener"` has near universal support:
![image](https://user-images.githubusercontent.com/10727862/153219844-b6b19e1d-b8e4-49f6-9753-17798b41fec8.png)
Besides, we're developers. We care about having the latest browser.

**Comment by Makonede** [comment]
+1
happy twosday, still no answer

**Comment by isobarbaric** [comment]
sadge still no answer 🐒

**Comment by be-favorite** [comment]
Still no answer..😭

**Comment by AlexWayfer** [comment]
> Still no answer..😭

It's not the official communication channel, there are no updates since 2009, so what do you expect?

**Comment by Makonede** [comment]
@AlexWayfer Haven't you been told multiple times to stop? GitHub has got to notice this at some point, and making these comments isn't helping.

**Comment by AlexWayfer** [comment]
> GitHub has got to notice this at some point, and making these comments isn't helping.

I can agree (that's why I'm subscribed to this issue), but I've been subscribed for a lot of years, and I suggest all of you to communicate with GitHub in other ways, rather this weird repo and issues.

You can see "Contact GitHub" and the site's footer. Any support request, [feedback](https://github.com/github/feedback/discussions) or even small tweet with mention will make more impact than a comment here, beleive me.

**Comment by eclairevoyant** [comment]
> A coworker of mine found a blog post (not from GitHub) which describes how `target="_blank"` can be a security and performance issue: https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/
> 
> If this is GitHub's rationale, great. It should be documented so this issue can be closed _after nine years_.

It's not a security issue if they enforce `rel=noopener`. But that's besides the point - this is the wrong place to comment, since this repo hasn't been updated in 14 years and the owner of this repo doesn't even work at Github anymore.

Complaining here is not a good use of time. You can either push for CommonMark to support attribute syntax or provide feedback / contact Github directly if you want to see any results.

**Comment by NOPR9D** [comment]
i tried to resume all method here :  
https://gist.github.com/NOPR9D/554cff1b13f0b01617d5564a514f454f

Exept use a external and custom api to make a redirection i had no solution. So we really need to develop an api just for that?

**Comment by 0xHady** [comment]
Somebody wake me up when this issue from 10 years ago is fixed

**Comment by rnsharma10** [comment]
Today, I came across this issue. Looks like we still have to wait for this feature to support.

**Comment by HudsonGraeme** [comment]
I opened a discussion [here](https://github.com/orgs/community/discussions/38634) on this issue. Hopefully a collaborator will have a look.

**Comment by vvaltchev** [comment]
@HudsonGraeme I believe they didn't "fix it" because it's a security issue, check here:
https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/

**Comment by HudsonGraeme** [comment]
> @HudsonGraeme I believe they didn't "fix it" because it's a security issue, check here: [blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security](https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/)

In 2022, `target="_blank"` is most assuredly not a security issue. Both `noreferrer` and `noopener` are supported by most browsers and effectively mitigate the security risk.

https://caniuse.com/rel-noreferrer
https://caniuse.com/rel-noopener

**Comment by MoElaSec** [comment]
so when is this 
```
[link](URL){:target="_blank" rel="noopener noreferrer"}
```
gonna be supported  🥲

**Comment by alimd** [comment]
never :cry:

**Comment by Holybasil** [comment]
still waiting

**Comment by saurabh-thecyberguy** [comment]
Happy new year 2023 Guys! 
11+ years of open issue :)

**Comment by kb0207** [comment]
damnit, still no updates

**Comment by choikangjae** [comment]
About 20% of the total population of the world is younger than this still opened issue 😂

**Comment by JamesAndersonJr** [comment]
> About 20% of the total population of the world is younger than this still opened issue 😂

🤔 Interesting perspective, indeed. Yet about 100% of GitHub engineers don't give a rat's arse.

**Comment by KTibow** [comment]
commenting here will:
- notify a hundred people

commenting here will not:
- make github care

**Comment by aravindvnair99** [comment]
+1 to @KTibow 

However, I also feel this is the wrong repository. @mojombo can you confirm?

The official discussion forum is here - https://github.com/orgs/community/discussions

**Comment by KTibow** [comment]
already opened suggestion: https://github.com/orgs/community/discussions/38634
i doubt mojombo will respond

**Comment by ankitkat042** [comment]
I often end up here looking for a solution for the same problem and realise that there is no solution for this till now, forget it and continue my work

**Comment by nexushoratio** [comment]
-1 to implementing this.  As a user, I absolutely hate it.  It is a pox on mankind.

**Comment by alimd** [comment]
> -1 to implementing this. As a user, I absolutely hate it. It is a pox on mankind.

-1 to you

**Comment by alan23273850** [comment]
@nexushoratio You did not mention why.

**Comment by nexushoratio** [comment]
> @nexushoratio You did not mention why.

As a user, I know how to open links in new windows/tabs.  Generally a simple middle-click.  But, *I* have a choice.  I may want to follow a link in the current window/tab.  It is not unusual for me to click a link to have it forcibly open in a new context.  So I then have to close the current one and navigate to the new one.  *My* choice was taken away from me.

There are few scenarios where opening a new window/tab by default makes sense.  Help on a form is probably the biggest.  And I don't think Markdown supports much in the way of forms.

**Comment by KTibow** [comment]
> My choice was taken away from me.

you can still open it in the current tab if you want:
- right-click > press l (copy link address) > press ctrl+l > press ctrl+v
- drag > drop onto address bar

**Comment by AlexWayfer** [comment]
> As a user, I know how to open links in new windows/tabs. Generally a simple middle-click. But, _I_ have a choice. I may want to follow a link in the current window/tab. It is not unusual for me to click a link to have it forcibly open in a new context. So I then have to close the current one and navigate to the new one. _My_ choice was taken away from me.

Probably, you should fill report to browsers. If they have support for middle mouse button or ctrl+lmb for forcing a new tab, probably they should have some key bindings for forcing the current tab, like alt-lmb, ctrl-mmb or something.

**Comment by shannaniggans** [comment]
Given the links i provide are to external tools to support the walkthroughs and courses, it is still annoying 13 years later ....

**Comment by MoElaSec** [comment]
#### **Happy New Year everyone it's "almost" 2024 🥳🎊** 

### Fun Facts: 
- This issue is 11 years+ old.

- The world population hit 8 billion his year (now at 8.1), which makes around 80 million people younger than this issue.

- 💀 <img src="https://en.wikipedia.org/wiki/Pyrenean_ibex" alt="Pyrenean ibex" style="max-width: 100%;"> was officially declared extinct this year. 

-

**Comment by MoElaSec** [comment]
#### **Happy New Year everyone it's "almost" 2024 🥳🎊** 

### Fun Facts: 
- This issue is 11 years+ old.

- The world population hit 8 billion his year (now at 8.1), which makes around 80 million people younger than this issue.

- 💀 <img src="https://en.wikipedia.org/wiki/Pyrenean_ibex" alt="Pyrenean ibex" style="max-width: 100%;"> was officially declared extinct this year. 

- GTA VI will be released before this issue is closed. (reminder today is the release date trailer 🙌) 

-  At least 152 Github users will be happy if this gets supported:
   ```markdown
   [link](URL){:target="_blank" rel="noopener noreferrer"}
   ```

**Comment by mhguelleh** [comment]
It's hard for me to believe GTA V came out after this issue and GTA VI will be released before this issue is ever fixed :exploding_head:

**Comment by ADV1K** [comment]
This has to be the oldest and most annoying issue I've ever seen.

I won't be surprised if I come back again searching for this next year.

**Comment by lnfel** [comment]
I'm leaving my mark in history (at least in this goddamn issue thread).

**Comment by sivarohithm** [comment]
Waiting 1 Million years

**Comment by prasku5** [comment]
I will check my comment in the next birth while this issue persists.

**Comment by p0tat0627** [comment]
Just to comment here for a reminder that, I did fix this issue from this thread just now.

**Comment by harshadsheelwant** [comment]
Doom-scrolling this issue is my new favorite pass time hobby.

**Comment by spookyahell** [comment]
I know this won't change it...
just wanted to be part of the people saying "Yes, I want this."

funny how I've been adding a target with no effect for years on GitHub. FML.

**Comment by brodrigz** [comment]
I want this!

**Comment by hi8esh** [comment]
Joining the "Never Ending **Waiting List**"

**Comment by Raghottamvkalkeri** [comment]
- [Python Official Documentation](https://docs.python.org/3/){:target="_blank"}

**Comment by alan23273850** [comment]
Has this issue been resolved yet? I can't wait anymore!

**Comment by HudsonGraeme** [comment]
> Has this issue been resolved yet? I can't wait anymore!

I don't want to call it too early, but this might be a wontfix situation

**Comment by TDroogers** [comment]
The fun part is, this is already possible and mentioned multiple times in this issue: ![demo](https://github.com)

`![test](https://github.com/mojombo/github-flavored-markdown/issues/28)`

**Comment by alan23273850** [comment]
@TDroogers I've just replaced your url with another YouTube link, and the popped up webpage just said "Non-Image content-type returned," so I think the solution is not complete enough...

**Comment by vpelss** [comment]
Odd that on MY readme.md I have no control... 
Github tells me what I want.
All hail github.

**Comment by alimd** [comment]
any news? 😆
